src/org/jitsi/service/neomedia/RTPEncoding.java

```diff
@@ -19,102 +19,45 @@
  * Inspired by the ORTC RTCRtpEncodingParameters, the RTPEncoding provides
  * RTP encoding related information.
  *
- * TODO this class needs to own the RemoteClock of this encoding.
- * TODO this class needs to know whether or not the RTP encoding is streaming.
- *
  * @author George Politis
  */
-public class RTPEncoding
+public interface RTPEncoding
 {
     /**
-     * Base simulcast stream quality order.
-     */
-    public static final int BASE_ORDER = 0;
-
-    /**
-     * The primary SSRC for this layering/encoding.
-     */
-    private final long primarySSRC;
-
-    /**
-     * The RTX SSRC for this layering/encoding.
-     */
-    private final long rtxSSRC;
-
-    /**
-     * The FEC SSRC for this layering/encoding.
-     */
-    private final long fecSSRC;
-
-    /**
-     * The {@link MediaStreamTrack} that owns this {@link RTPEncoding}.
-     */
-    private final MediaStreamTrack mediaStreamTrack;
-
-    /**
-     * Ctor.
-     *
-     * @param mediaStreamTrack
-     * @param primarySSRC
-     * @param rtxSSRC
-     * @param fecSSRC
-     */
-    RTPEncoding(MediaStreamTrack mediaStreamTrack,
-                       long primarySSRC, long rtxSSRC, long fecSSRC)
-    {
-        this.mediaStreamTrack = mediaStreamTrack;
-        this.primarySSRC = primarySSRC;
-        this.rtxSSRC = rtxSSRC;
-        this.fecSSRC = fecSSRC;
-    }
-
-    /**
-     * Gets the {@link MediaStreamTrack} that owns this {@link RTPEncoding}.
-     *
-     * @return the {@link MediaStreamTrack} that owns this {@link RTPEncoding}.
-     */
-    public MediaStreamTrack getMediaStreamTrack()
-    {
-        return mediaStreamTrack;
-    }
-
-    /**
      * Gets the primary SSRC for this layering/encoding.
      *
      * @return the primary SSRC for this layering/encoding.
      */
-    public long getPrimarySSRC()
-    {
-        return primarySSRC;
-    }
+    long getPrimarySSRC();
 
     /**
      * Gets the RTX SSRC for this layering/encoding.
      *
      * @return the RTX SSRC for this layering/encoding.
      */
-    public long getRTXSSRC()
-    {
-        return rtxSSRC;
-    }
+    long getRTXSSRC();
 
     /**
-     * Gets the FEC SSRC for this layering/encoding.
+     * Gets a boolean value indicating whether or not this instance is
+     * streaming.
      *
-     * @return the RTX SSRC for this layering/encoding.
+     * @return true if this instance is streaming, false otherwise.
+     */
+    boolean isActive();
+
+    /**
+     * Gets the {@link MediaStreamTrack} that this instance belongs to.
+     *
+     * @return the {@link MediaStreamTrack} that this instance belongs to.
      */
-    public long getFECSSRC()
-    {
-        return fecSSRC;
-    }
+    MediaStreamTrack getMediaStreamTrack();
 
     /**
      *
-     * @param ssrc
+     * @param buf
```

Comment:
Can we have some docs here, please? The implementation references this, and I don't think the meaning is obvious.